R/plot.R

```diff
@@ -0,0 +1,449 @@
+    y_lab <- "Log(Density Ratio)"
+
+  } else {
+    y_lab <- "Density Ratio"
+  }
+
+
+  if(output == "individual"){
+  # Create list storage for plots object (for iteration)
+  plots <- list()
+  for(var in vars){
+    plots[[var]] <- individual_uni_plot(data, var, y_lab, sample.facet)
+  }
+  return(plots)
+
+  }
+
+  if (output == "assembled"){
+    data <- data %>%
+      pivot_longer(cols = vars,
+                   names_to = "variable",
+                   values_to = "value")
+
+    # Maximum scale for y
+    y_max <- max(1,data$dr)
+    y_min <- min(-1, data$dr)
+
+    plot <- ggplot(data) +
+      geom_point(aes(x = value, y = dr, col = sample),
+                 alpha = .6) +
+      theme_bw() +
+      labs(title = "Scatter plot of individual values and density ratio",
+           color = "Sample",
+           y = "Density ratio") +
+      scale_color_manual(values = c("firebrick", "steelblue"),
+                         breaks = c("numerator", "denominator"),
+                         labels = c("Numerator", "Denominator")) +
+      scale_y_continuous(limits = c(y_min, y_max))
+
+    if(sample.facet){
+      plot <- plot +
+        facet_grid(cols = vars(sample),
+                   rows = vars(variable),
+                   scales = "free_x")
+
+      } else {
+        plot <- plot +
+          facet_wrap(~variable, scales = "free_x", nrow = nrow)
+    }
+
+  }
+  return(plot)
+
+}
+
+
+#' Bivariate plot
+#'
+#' @inheritParams individual_uni_plot
+#' @param show.sample Logical indicating whether to give different shapes to observations, depending on the sample they come from (numerator or denominator). Defaults to FALSE.
+#'
+#' @return Bivariate plot
+#'
+#' @examples
+individual_biv_plot <- function(data, vars, logscale, show.sample){
+
+  dr_max <- ifelse(logscale, max(2, data$dr), max(exp(2), data$dr))
+  dr_min <- ifelse(logscale, min(-2, data$dr), min(exp(-2), data$dr))
+
+  plot <-
+    ggplot(data, mapping = aes(x = .data[[vars[1]]], y = .data[[vars[2]]])) +
+    geom_point(aes(colour = dr, shape = if(show.sample) sample else NULL)) +
+    scale_colour_gradient2(low = "firebrick",
+                           high = "steelblue",
+                           mid = "lightyellow",
+                           limits = c(dr_min, dr_max)) +
+    theme_bw() +
+    labs(title = "Scatter plot, with density ratio mapped to colour",
+         colour = "Log (Density ratio)") +
+    scale_shape_manual(values = c(21, 24))
+
+  return(plot)
+}
+
+#' Densityratio in bidimensional plot
+#'
+#' Plots a scatterplot of two variables, with densityratio mapped to the colour scale.
+#'
+#' @inheritParams plot.univariate
+#' @inheritParams individual_biv_plot
+#'
+#' @return Scatter plot of two variables, with density ratio mapped to the colour scale.
+#' @export
+#'
+#' @examples
+plot_bivariate <- function(object, vars, samples = "both",
+                           output = "assembled", logscale = TRUE, show.sample = FALSE) {
```

Comment:
Did that! I also added one check so that if the length of vars1 == 1 and vars2 == 1, then it defaults to grid = FALSE (A grid makes no sense for one plot). And another one so that if length(vars1) and &is.nul(vars2), then there's an error (youneed to variables for a bivariate plot).